test/jest/__tests__/components/evm/AddressQR.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { shallowMount, VueWrapper } from '@vue/test-utils';
+import AddressQR from 'components/evm/AddressQR.vue';
+import QRious from 'qrious';
+
+const SIZE_A = 400;
+const ADDRESS_A = '0x1234567890123456789012345678901234567890';
+const ADDRESS_B = '0x0000000000000000000000000000000000000000';
+const BACKGROUND_COLOR = 'rgb(0, 0, 0)';
+
+jest.mock('qrious', () => jest.fn());
+
+Object.defineProperty(window, 'getComputedStyle', {
+    value: () => ({
+        getPropertyValue: () => BACKGROUND_COLOR,
+    }),
+});
+
+interface WrapperVM {$nextTick: ()=>Promise<void>}
+
+describe('AddressQR.vue', () => {
+    const getNewWrapper = () => shallowMount(AddressQR, {
+        props: {
+            address: ADDRESS_A,
+            size: SIZE_A,
+        },
+        attachTo: document.body,
+        global: {
+            mocks: {
+                $style: {
+                    '--header-bg-color': BACKGROUND_COLOR,
+                },
+            },
+        },
+    }) as VueWrapper<never>;
+
+    let wrapper: VueWrapper<never>;
+
+    beforeEach(() => {
+        wrapper = getNewWrapper();
+    });
+
+    afterEach(() => {
+        wrapper.unmount();
+        (QRious as jest.Mock).mockReset();
+    });
+
+    it('should render correctly', async () => {
+        expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should have the correct name', () => {
+        expect(AddressQR.name).toBe('AddressQR');
+    });
+
+    it('should render a canvas element with the id "qr-code"', () => {
+        expect(wrapper.find('#qr-code').exists()).toBe(true);
+    });
+
+    it('should call QRious with the correct options', () => {
+        expect(QRious).toHaveBeenCalledWith({
+            background: BACKGROUND_COLOR,
+            level: 'H',
+            size: SIZE_A,
+            element: expect.any(HTMLElement),
+            value: ADDRESS_A,
+        });
+    });
+
+    it('should update the QR code when the address prop is changed', async () => {
+        wrapper.setProps({ address: ADDRESS_B });
+        await (wrapper.vm as WrapperVM).$nextTick();
+        expect(QRious).toHaveBeenCalledWith({
+            background: BACKGROUND_COLOR,
+            level: 'H',
+            size: SIZE_A,
+            element: expect.any(HTMLElement),
+            value: ADDRESS_B,
+        });
+    });
+
+
```

Comment:
if possible I would like a test that says this qr code does translate to the address input, and maybe test a couple of different random addresses.

Reply:
In the current tests we have the library code has not been used at all because it's mocked. We just can see if it was called, and how many times and see the parameters, but we do nothing with that call. Therefore, the QR does not get created at all on these tests.

The task you request will require installing a canvas package to have canvas support on testing to capture the actual QR draw and then compare it to another previously saved trusted draw. 

I would suggest breaking this part [into another issue](https://github.com/telosnetwork/telos-wallet/issues/269).